src/main/java/org/hisp/dhis/Dhis2.java

```diff
@@ -2454,4 +2466,55 @@ public List<JobNotification> getJobNotifications(JobCategory category, String id
 
     return new ArrayList<>(Arrays.asList(response));
   }
+
+  /**
+   * Retrieves a {@link FileResource}.
+   *
+   * @param id the object identifier.
+   * @return the {@link FileResource}.
+   * @throws Dhis2ClientException if the object does not exist.
+   */
+  public FileResource getFileResource(String id) {
+    return getObject(
+        config.getResolvedUriBuilder().appendPath("fileResources").appendPath(id),
+        Query.instance(),
+        FileResource.class);
+  }
+
+  /**
+   * Retrieves the validation results of a {@link DataSet} in a particular {@link Period} and {@link
+   * OrgUnit}.
+   *
+   * @param query the {@link DataSetValidationQuery}.
+   * @return The validation results of the {@link DataSet} in the specified {@link Period} and
+   *     {@link OrgUnit}.
+   */
+  public Validation getDataSetValidation(DataSetValidationQuery query) {
+    return getDataSetValidationResponse(
+        config
+            .getResolvedUriBuilder()
+            .appendPath("validation/dataSet")
+            .appendPath(query.getDataSet())
+            .addParameter(FIELDS_PARAM, VALIDATION_RULES_FIELDS),
+        query,
+        Validation.class);
+  }
+
+  /**
+   * Retrieves a list of {@link ValidationRule}.
+   *
+   * @param dataSet the object identifier of the Data Set where the validations apply.
+   * @return list of {@link ValidationRule}.
+   */
+  public List<ValidationRule> getValidationRules(String dataSet) {
+    return getObject(
+            config
+                .getResolvedUriBuilder()
+                .appendPath("validationRules")
+                .addParameter(DATA_SET_PARAM, dataSet)
+                .addParameter(FIELDS_PARAM, "*"),
```

Comment:
Let's not use wildcard `*` and instead use explicit fields, that will reduce the response payload and make it consistent with other methods.